app/components/common/paperItem/collectionButton.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from "react";
+import { withStyles } from "../../../helpers/withStylesHelper";
+import GlobalDialogManager from "../../../helpers/globalDialogManager";
+import { trackEvent } from "../../../helpers/handleGA";
+import ActionTicketManager from "../../../helpers/actionTicketManager";
+import Icon from "../../../icons";
+const styles = require("./collectionButton.scss");
+
+interface CollectionButtonProps {
+  paperId: number;
+  pageType: Scinapse.ActionTicket.PageType;
+  hasCollection: boolean;
+  actionArea?: Scinapse.ActionTicket.ActionArea;
+  onRemove?: (paperId: number) => Promise<void>;
+}
+
+const CollectionButton: React.SFC<CollectionButtonProps> = ({
+  paperId,
```

Comment:
Why did you take out all the elements and declare them? Is it because declarations such as props.onRemove are too long?

Reply:
just habit lol.